### PR TITLE
fix(pagination): handle page size zero

### PR DESCRIPTION
**Pull request by akashsonune**
Inconsistent state when pagesize is zero

---

[Documentation](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/).
[Examples](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/element-examples/).
[Dashboards Demo](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/dashboards-demo/).
[Playwright report](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/playwright-report/).

**Coverage Reports:**

Code Coverage

- [charts-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/charts-ng/)
- [dashboards-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/dashboards-ng/)
- [element-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/element-ng/)
- [element-translate-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/element-translate-ng/)
- [maps-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/maps-ng/)
- [native-charts-ng](https://d33c9dcnqinn2a.cloudfront.net/pr-1877/pages/coverage/native-charts-ng/)

### CODE REVIEW

**Comment by ljanner** [comment]
Seems like we lack test coverage for the pagination component. 

Can we either add more test coverage here or otherwise create an issue so we are aware and have it on the backlog.  The component has many edge cases, which are not covered by tests. For example `currentPage` > `totalPages`

File: projects/element-ng/pagination/si-pagination.component.ts [hunk 1]
```diff
@@ -68,9 +68,11 @@ export class SiPaginationComponent {
    */
   readonly navAriaLabel = input(t(() => $localize`:@@SI_PAGINATION.NAV_LABEL:Pagination`));
 
-  private readonly calculatedTotalPages = computed(
-    () => this.totalPages() ?? Math.ceil((this.totalRowCount() ?? 0) / (this.pageSize() ?? 1))
-  );
+  private readonly calculatedTotalPages = computed(() => {
+    const pageSize = this.pageSize();
+    const safePageSize = pageSize && pageSize > 0 ? pageSize : 1;
```
Comment (spike-rabbit) [inline projects/element-ng/pagination/si-pagination.component.ts]:
```suggestion
    const safePageSize = Math.max(pageSize ?? 0, 1)
```

**Review by spike-rabbit — CHANGES_REQUESTED** [review]
please add unit test

File: projects/element-ng/pagination/si-pagination.component.ts (hunk 1 above)
Comment (akashsonune) [reply projects/element-ng/pagination/si-pagination.component.ts]: 👍

**Comment by akashsonune** [comment]
@spike-rabbit can you check

**Review by spike-rabbit — APPROVED** [review]
